common/djangoapps/dark_lang/tests.py

```diff
@@ -229,78 +226,126 @@ def test_partial_match_es_la(self, latin_america_code):
+        self._set_client_session_language('notrel')
+        self._post_set_preview_lang('rel')
+        self.client.get('/home')
+        self.assert_session_lang_equals(
             'rel',
-            self.process_request(preview_lang='rel', language_session_key='notrel')
+            self.client.session
         )
 
     def test_preview_lang_with_dark_language(self):
-        self.assertSessionLangEquals(
+        self._post_set_preview_lang('unrel')
+        self.client.get('/home')
+        self.assert_session_lang_equals(
             'unrel',
-            self.process_request(preview_lang='unrel')
+            self.client.session
         )
 
-        self.assertSessionLangEquals(
+        # Test a clear and then a set of the preview language
+        self._post_clear_preview_lang()
+        self._post_set_preview_lang('unrel')
+        self.client.get('/home')
+        self.assert_session_lang_equals(
             'unrel',
-            self.process_request(preview_lang='unrel', language_session_key='notrel')
+            self.client.session
         )
 
-    def test_clear_lang(self):
-        self.assertSessionLangEquals(
-            UNSET,
-            self.process_request(clear_lang=True)
+    def test_empty_preview_language(self):
+        # When posting an empty preview_language the currently set language should not change
+        self._set_client_session_language('rel')
+        self._post_set_preview_lang(' ')
+        self.client.get('/home')
+        self.assert_session_lang_equals(
+            'rel',
+            self.client.session
         )
 
-        self.assertSessionLangEquals(
+    def test_clear_lang(self):
+        # Clear a language when no language was set
+        self._post_clear_preview_lang()
+        self.client.get('/home')
+        self.assert_session_lang_equals(
             UNSET,
-            self.process_request(clear_lang=True, language_session_key='rel')
+            self.client.session
         )
 
-        self.assertSessionLangEquals(
+        # Set a language and clear it to ensure the clear is working as expected
+        self._post_set_preview_lang('notclear')
+        self._post_clear_preview_lang()
+        self.client.get('/home')
+        self.assert_session_lang_equals(
             UNSET,
-            self.process_request(clear_lang=True, language_session_key='unrel')
+            self.client.session
         )
 
     def test_disabled(self):
         DarkLangConfig(enabled=False, changed_by=self.user).save()
 
         self.assertAcceptEquals(
             'notrel;q=0.3, rel;q=1.0, unrel;q=0.5',
-            self.process_request(accept='notrel;q=0.3, rel;q=1.0, unrel;q=0.5')
+            self.process_middleware_request(accept='notrel;q=0.3, rel;q=1.0, unrel;q=0.5')
         )
 
-        self.assertSessionLangEquals(
+        # With DarkLang disabled the clear should not change the session language
+        self._post_clear_preview_lang()
```

Comment:
I'm not following this test. The clear is done before _set_client_session_language, so why would the clear persist anyway?